### DISCUSSION TITLE
How to play a marker? from "tm" for "dr"

### DISCUSSION THREAD

**Discussion by ArtemMikhaylov** [airbnb/lottie-ios#1642] [Q&A] [answered]
Hello! I have some troubles working with iOS lottie api.

In Android I can play a marker by its name, without knowing is there some other markers, which marker goes next, or if this marker is the last one!  Just like that: 

```java
animationView.setRepeatCount(LottieDrawable.INFINITE);
animationView.setMinAndMaxFrame("surprise");
animationView.playAnimation();
```
And I can't find the way in iOS to do so. Isn't that what `dr` parameter in marker object for?
```json
{"tm":80,"cm":"surprise","dr":44}
```

And even without method `setMinAndMaxFrame("surprise")` in android I still can get the Marker object itself and make it work:

```java
Marker marker = animationView.getComposition().getMarker("surprise");
float dr = marker.durationFrames;
float tm = marker.startFrame;
```
But I don't see how to get Marker object on iOS either.

Or am I missing something?

**Comment by calda** [comment, marked as answer]
It looks like markers with durations (`dr` values) are currently unsupported on iOS -- we don't parse the `dr` field. Open to adding support for this, we just need somebody to implement it.

If you have two markers, each with their own `tm` value, you can use `AnimationView.play(fromMarker:toMarker:)`. This is how `AnimatedSwitch` and `AnimatedButton` work, for example.

**Comment by kmcbride** [comment]
This should now be possible as of https://github.com/airbnb/lottie-ios/pull/1749
